util/settings/settings.go

```diff
@@ -341,7 +345,25 @@ func (mgr *SettingsManager) GetKustomizeBuildOptions() (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return argoCDCM.Data[kustomizeBuildOptionsKey], nil
+	if value, ok := argoCDCM.Data[kustomizeBuildOptionsKey]; ok {
+		return value, nil
+	}
+	return "", nil
+}
+
+// GetHelmDirectoryEnforcerLevel loads the helm directory enforcer level from argocd-cm ConfigMap
+func (mgr *SettingsManager) GetHelmDirectoryEnforcerLevel() (v1alpha1.HelmDirectoryEnforcerLevel, error) {
+	argoCDCM, err := mgr.getConfigMap()
+	if err != nil {
+		return "", err
+	}
+	enforcerLevel := v1alpha1.HelmDirectoryEnforcerLevel(argoCDCM.Data[helmDirectoryEnforcerLevelKey])
+	switch enforcerLevel {
+	case v1alpha1.EnforcerLevelRepo, v1alpha1.EnforcerLevelStrict:
+		return enforcerLevel, nil
+	default:
+		return v1alpha1.EnforcerLevelRepo, nil
```

Comment:
Hm, same thought as earlier: Should we just ignore invalid values, or error out? I'm thinking of typos, for example.

Reply:
Will also change